### PR TITLE
BNN model and update requirements

### PR DESCRIPTION
**Pull request by dounia4112**
### BnnModel File

- 	Creates the Bayesiane neural network layer (BnnLayer Class).

- 	Initializes a Bayesian neural network model with two layers (BayesianModel Class).
- 	Implements the forward method using ReLU activation function.

### train_bnn File
**Install dependencies :** 

- 	torch, scikit-learn, matplotlib (added to requirements.txt).

**Data Preparation:**

- 	Loads the data and splits it into training and test sets.

**Tensor Conversion:**

- 	Converts NumPy arrays to PyTorch tensors for both training and test sets.

**Model Initialization:**

- 	Defines a Bayesian neural network model (BayesianModel) with one hidden layer (BnnLayer) having 20 units.
- 	Sets up the loss function as Mean Squared Error (MSE) and the Adam optimizer.

**Training Loop:**

- 	Executes a training loop for 1000 epochs.
- 	Performs forward and backward passes, optimizing the model parameters to minimize the MSE loss.
- 	Records and stores the training loss at each epoch.

### CODE REVIEW

**Comment by 1hachem** [comment]
please you also need to work on a the following task and document how we can run it on the available datasets  
- [ ] fit BNN : fit bnn to provided data and generate plots.

**Comment by 1hachem** [comment]
well done Dounya 👍🏼, we will contact you soon for an interview inchalah
